### PR TITLE
Add support for loading custom patterns

### PR DESCRIPTION
**Pull request by shikharmittal**
#15

### CODE REVIEW

**Comment by shikharmittal** [comment]
@Marusyk - I have tried to make some changes to add support for loading custom patterns. Please review and let me know your feedback.

File: src/Grok.Net/Grok.cs [hunk 1]
```diff
@@ -123,10 +124,21 @@ private void LoadPatterns()
                             ProcessPatternLine(sr.ReadLine());
                         }
                     }
+                }
+            }
 
-                    _patternsLoaded = true;
+            if (File.Exists(customPatternFileName))
```
Comment (Marusyk) [inline src/Grok.Net/Grok.cs:130]: I would move it to `LoadCustomPatterns()`. What do you think?

File: src/Grok.Net/Grok.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace GrokNet
 {
     public class Grok
     {
+        private const string customPatternFileName = "grok-custom-patterns";
```
Comment (Marusyk) [inline src/Grok.Net/Grok.cs:13]: Can we avoid of defining the file name and follow this: 
https://github.com/Marusyk/grok.net/issues/15#issuecomment-860511754 ?

**Review by Marusyk — CHANGES_REQUESTED** [review]

File: src/Grok.Net/Grok.cs (hunk 2 above)
Comment (shikharmittal) [reply src/Grok.Net/Grok.cs:13]: So we look for a directory named 'patterns' and load all the files present inside that?

Comment (Marusyk) [reply src/Grok.Net/Grok.cs:13]: Correct

Comment (Marusyk) [reply src/Grok.Net/Grok.cs:13]: Could you please remove this const? It seems we don't  neet it anymore

**Review by Marusyk — CHANGES_REQUESTED** [review]
LGTM 👍 . Please fix one small comment

**Review by Marusyk — APPROVED** [review]
Many thanks for your contribution

**Pull request merged**